### PR TITLE
Generic versions of resolveReference

### PR DESCRIPTION
**Pull request by msladek**
Added generic WebUtilsService.resolveReference(...)

### CODE REVIEW

File: src/main/java/com/celements/web/service/WebUtilsService.java [hunk 1]
```diff
@@ -470,22 +500,39 @@ public EntityReference resolveEntityReference(String name, EntityType type) {
...
-    return ref;
+    return resolveReference(name, ENTITY_TYPE_MAP.inverse().get(type), wikiRef);
   }
 
   @Override
   public EntityReference resolveRelativeEntityReference(String name, EntityType type) {
-    EntityReference ref = refResolver_relative.resolve(name, type);
-    _LOGGER.debug("resolveRelativeEntityReference: for [" + name + "] got reference [" + ref + "]");
+    EntityReference ref = relativeRefResolver.resolve(name, type);
+    LOGGER.debug("resolveRelativeEntityReference: for [" + name + "] got reference [" + ref + "]");
```
Comment (fpichler) [inline src/main/java/com/celements/web/service/WebUtilsService.java]: please use logger with StringBuilder call debug(String message, Object replacement1..., Throwable)

File: src/main/java/com/celements/web/service/IWebUtilsService.java [hunk 2]
```diff
@@ -138,6 +143,12 @@ public String getDocSection(String regex, DocumentReference docRef, int section)
   public EntityReference resolveEntityReference(String name, EntityType type,
       WikiReference wikiRef);
 
+  public <T extends EntityReference> T resolveReference(@NotNull String name,
```
Comment (fpichler) [inline src/main/java/com/celements/web/service/IWebUtilsService.java]: please add JavaDoc explaining shortly the usage and hint to resolveSpaceReference, resolveDocumentReference, resolveAttachmentReference and resolveWikiReference

**Comment by fpichler** [comment]
Please add a CELDEV issue in Jira explaining shortly the change. Reference the JIRA-Issue in the pull request description.

**Comment by msladek** [comment]
https://synjira.atlassian.net/browse/CELDEV-251

**Pull request merged**